src/main/java/Circle.java

```diff
@@ -0,0 +1,17 @@
+public class Circle extends Shape {
+    private final double radius;
+    private final double pi = Math.PI;
```

Comment:
stałe wielkimi literami. To może być 'private static final'.

Ale z drugiej strony - czy to faktycznie potrzebne przypisanie? :) Czy użycie Math.PI bezpośrednio w perimeter() nie wystarczy? ;)